### PR TITLE
bool values defaulting to true were always true

### PR DESCRIPTION
**Pull request by mohit**
set the default in the generated flagSet to be the incoming value set in the field (by default bools are false). Always use the value from the flagset

### CODE REVIEW

**Comment by mohit** [comment]
@natebrennand ran into a rather curious bug with bools. Any ideas on how to deal with it. I added a test that was failing before my change.

**Comment by natebrennand** [comment]
Weird... when I checkout https://github.com/Clever/configure/pull/9/commits/252ae52c3e315d62a9535daabc475d72ef5d736f and run it locally, the test _does_ pass. Is this properly testing the case that you were encountering?

File: configure.go [hunk 1]
```diff
@@ -116,8 +117,11 @@ func Configure(configStruct interface{}) error {
 				valueField.SetString(*flagStringValueMap[tagVal])
 			}
 		case reflect.Bool:
-			if *flagBoolValueMap[tagVal] {
-				flagFound = true
+			if flagBoolValueMap[tagVal] != nil {
```
Comment (natebrennand) [inline configure.go]: 🤦‍♂️ good catch

File: Makefile [hunk 2]
```diff
@@ -6,7 +6,7 @@ PKG := github.com/Clever/configure
 PKGS := $(shell go list ./... | grep -v example)
 .PHONY: all test
 
-$(eval $(call golang-version-check,1.8))
+$(eval $(call golang-version-check,1.9))
```
Comment (natebrennand) [inline Makefile]: Can you leave this at 1.8 until we upgrade everything to 1.9? The golang.mk will prevent the versioning error from affecting you

File: circle.yml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 machine:
   post:
   - cd $HOME && git clone --depth 1 -v git@github.com:clever/ci-scripts.git && cd ci-scripts && git show --oneline -s
-  - $HOME/ci-scripts/circleci/golang-install 1.8
+  - $HOME/ci-scripts/circleci/golang-install 1.9
```
Comment (natebrennand) [inline circle.yml]: same w/ this at 1.8

File: configure.go [hunk 4]
```diff
@@ -78,7 +78,7 @@ func Configure(configStruct interface{}) error {
 			return ErrNotReference
 		}
 
-		// currently we only support strings
+		// currently we only support strings and bools
```
Comment (natebrennand) [inline configure.go:81]: 👌

File: configure.go [hunk 5]
```diff
@@ -116,8 +117,11 @@ func Configure(configStruct interface{}) error {
 				valueField.SetString(*flagStringValueMap[tagVal])
 			}
 		case reflect.Bool:
-			if *flagBoolValueMap[tagVal] {
-				flagFound = true
+			if flagBoolValueMap[tagVal] != nil {
+				// we can only know if a bool flag was set if the default was changed
+				if *flagBoolValueMap[tagVal] != config.Field(i).Bool() {
```
Comment (natebrennand) [inline configure.go]: Should we just fold these expressions into the one `if` boolean expression?

Comment (mohit) [reply configure.go]: https://github.com/Clever/configure/pull/9/commits/016083ff03437b4ecef77fdfe91fa20c9c8168e0

**Review by natebrennand — APPROVED** [review]
Would be nice to rebase or squash some of these commits.
Thanks for the bug fix! 💯

**Comment by mohit** [comment]
squashed existing commits and added a new one for GitHub releases.

**Pull request merged**